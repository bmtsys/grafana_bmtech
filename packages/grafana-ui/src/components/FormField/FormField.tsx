import React, { InputHTMLAttributes, FunctionComponent } from 'react';
import { FormLabel } from '..';

export interface Props extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  labelWidth?: number;
  inputWidth?: number;
}

const defaultProps = {
  labelWidth: 6,
  inputWidth: 12,
};

const FormField: FunctionComponent<Props> = ({ label, labelWidth, inputWidth, ...inputProps }) => {
  return (
    <div className="form-field">
      <FormLabel width={labelWidth}>{label}</FormLabel>
      <input type="text" className={`gf-form-input width-${inputWidth}`} {...inputProps} />
    </div>
  );
};

FormField.defaultProps = defaultProps;
export { FormField };
