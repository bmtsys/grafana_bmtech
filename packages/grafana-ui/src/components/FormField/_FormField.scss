.form-field {
  margin-bottom: $gf-form-margin;
  display: flex;
  flex-direction: row;
  align-items: center;
  text-align: left;
  position: relative;

  &--grow {
    flex-grow: 1;
  }
}
