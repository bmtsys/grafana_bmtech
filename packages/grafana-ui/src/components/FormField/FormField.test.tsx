import React from 'react';
import { shallow } from 'enzyme';
import { FormField, Props } from './FormField';

const setup = (propOverrides?: object) => {
  const props: Props = {
    label: 'Test',
    labelWidth: 11,
    value: 10,
    onChange: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<FormField {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
