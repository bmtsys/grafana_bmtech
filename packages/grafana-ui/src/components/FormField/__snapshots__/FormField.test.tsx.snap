// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="form-field"
>
  <Component
    width={11}
  >
    Test
  </Component>
  <input
    className="gf-form-input width-12"
    onChange={[MockFunction]}
    type="text"
    value={10}
  />
</div>
`;
